packages/p2p-v2/src/pages/orders/components/ChatFooter/ChatFooter.tsx

```diff
@@ -20,18 +24,74 @@ const ChatFooter = ({ isClosed }: TChatFooterProps) => {
             </div>
         );
     }
+
+    const sendChatMessage = () => {
+        const elTarget = textInputRef.current;
+        const shouldRestoreFocus = document.activeElement === elTarget;
+
+        if (elTarget?.value) {
+            sendMessage(elTarget.value);
+            elTarget.value = '';
+            setValue('');
+
+            if (shouldRestoreFocus) {
+                elTarget.focus();
+            }
+        }
+    };
+
+    const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+        if (event.key === 'Enter' && !isMobile) {
+            if (event.ctrlKey || event.metaKey) {
+                const element = event.target as HTMLInputElement;
+                const { value } = element;
+
+                if (typeof element.selectionStart === 'number' && typeof element.selectionEnd === 'number') {
+                    element.value = `${value.slice(0, element.selectionStart)}\n${value.slice(element.selectionEnd)}`;
+                } else if (document.selection?.createRange) {
+                    element.focus();
+
+                    const range = document.selection.createRange();
+
+                    range.text = '\r\n';
+                    range.collapse(false);
+                    range.select();
+                }
+            } else {
+                event.preventDefault();
+                sendChatMessage();
+            }
+        }
+    };
+
     return (
-        <div className='px-[2.4rem] pt-[1.6rem] pb-[2.8rem]'>
+        <div className='px-[2.4rem] pt-[1.6rem] pb-[2.8rem] w-full'>
+            {/* TODO: replace with text area component once multiline functionality available */}
             <Input
                 isFullWidth
                 label='Enter message'
                 maxLength={5000}
                 message={`${value.length}/5000`}
                 onChange={onChange}
-                //TODO: handle send message
-                rightPlaceholder={<ChatFooterIcon length={value.length} onClick={() => undefined} />}
+                onKeyDown={handleKeyDown}
+                ref={ref => (textInputRef.current = ref)}
+                rightPlaceholder={
+                    <ChatFooterIcon
+                        length={value.length}
+                        onClick={() => (value.length > 0 ? sendChatMessage() : fileInputRef.current?.click())}
+                    />
+                }
                 wrapperClassName='p2p-v2-chat-footer__input'
             />
+            <div className='hidden'>
+                <Input
+                    data-testid='dt_p2p_v2_file_input'
+                    name='file'
+                    onChange={e => sendFile(e.target.files?.[0] as File)}
```

Comment:
@nada-deriv, since we would only want to invoke  `sendFile` when the `e.target.files?.[0]` is of type `File`, you should use type narrowing here
```suggestion
                    onChange={e => {
                        if (e.target.files?.[0]) {
                            sendFile(e.target.files[0]);
                        }
                    }}
```